buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -802,6 +802,7 @@ class BeamModulePlugin implements Plugin<Project> {
             project.configurations.shadow.artifacts.files.each {
               FileTree exposedClasses = project.zipTree(it).matching {
                 include "**/*.class"
+                exclude "META-INF/**" // BEAM-5919: Exclude module-info.class for Java 9 build support
```

Comment:
I feel like it will be a long while before "shading" tools support multi-release jars and can see the amount of confusion/bugs people attempting to shade these will cause.

Reply:
I decided to use a single `*` for the version directory instead of `**`.
Since `**` matches any number of directories as per:
https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/util/PatternFilterable.html